### DISCUSSION TITLE
Sharding on top of etcd

### DISCUSSION THREAD

**Discussion by Tochemey** [etcd-io/etcd#16199] [Q&A] [answered]
Hello folks,

I would like to know whether it is possible to implement a data sharding mechanism on top of etcd and whether there are some guidelines to follow to properly achieve that.

Thanks

**Comment by jmhbnz** [comment, marked as answer]
Thanks for your question @Tochemey.   To the best of my knowledge this is not possible with a standard etcd cluster deployment however there are some third party components you can put infront for rudimentary sharding, refer:
- https://github.com/sharding-db/etcd-sharding-proxy.
- https://github.com/flaviostutz/etcd-hashring

Please note I have not tested these projects and they are not supported by the etcd project. Use at your own risk.

There is a very brief mention on the etcd stance on sharding in our docs here: https://etcd.io/docs/v3.5/learning/why/#newsql-cloud-spanner-cockroachdb-tidb
